codec-http/src/main/java/io/netty/handler/codec/http/HttpServerCodec.java

```diff
@@ -69,4 +79,41 @@ public HttpServerCodec(int maxInitialLineLength, int maxHeaderSize, int maxChunk
     public void upgradeFrom(ChannelHandlerContext ctx) {
         ctx.pipeline().remove(this);
     }
+
+    private final class HttpServerRequestDecoder extends HttpRequestDecoder {
+        public HttpServerRequestDecoder(int maxInitialLineLength, int maxHeaderSize, int maxChunkSize) {
+            super(maxInitialLineLength, maxHeaderSize, maxChunkSize);
+        }
+
+        public HttpServerRequestDecoder(int maxInitialLineLength, int maxHeaderSize, int maxChunkSize,
+                                        boolean validateHeaders) {
+            super(maxInitialLineLength, maxHeaderSize, maxChunkSize, validateHeaders);
+        }
+
+        public HttpServerRequestDecoder(int maxInitialLineLength, int maxHeaderSize, int maxChunkSize,
+                                        boolean validateHeaders, int initialBufferSize) {
+            super(maxInitialLineLength, maxHeaderSize, maxChunkSize, validateHeaders, initialBufferSize);
+        }
+
+        @Override
+        protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> out) throws Exception {
+            int oldSize = out.size();
+            super.decode(ctx, buffer, out);
+            int size = out.size();
+            for (int i = oldSize; i < size; i++) {
```

Comment:
the `List` interface allows base classes to do some interesting things which may make this iteration not do exactly what we want. Maybe its OK to assume the super class behavior only adds to the list, and doesn't change indexes though bcz its an internal class? This method is also used in HttpClientCodec too. We have discussed before but it seems to remove ambiguity and overhead getting rid of the list interface (for future releases).